{"name": "Применение одного промокода дважды", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.ElementClickInterceptedException: Message: element click intercepted: Element <button type=\"submit\" class=\"button\" name=\"apply_coupon\" value=\"...\">Применить купон</button> is not clickable at point (509, 282). Other element would receive the click: <div class=\"inner\">...</div>\n  (Session info: chrome=138.0.7204.183); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#elementclickinterceptedexception\nStacktrace:\n#0 0x5fa8024e92ca <unknown>\n#1 0x5fa801f90550 <unknown>\n#2 0x5fa801fe8f5c <unknown>\n#3 0x5fa801fe6dc2 <unknown>\n#4 0x5fa801fe4462 <unknown>\n#5 0x5fa801fe3b80 <unknown>\n#6 0x5fa801fd62ea <unknown>\n#7 0x5fa802007bc2 <unknown>\n#8 0x5fa801fd5c6a <unknown>\n#9 0x5fa802007d8e <unknown>\n#10 0x5fa80202d9e6 <unknown>\n#11 0x5fa802007993 <unknown>\n#12 0x5fa801fd3d6b <unknown>\n#13 0x5fa801fd5141 <unknown>\n#14 0x5fa8024ae2ab <unknown>\n#15 0x5fa8024b20b9 <unknown>\n#16 0x5fa802495139 <unknown>\n#17 0x5fa8024b2c68 <unknown>\n#18 0x5fa80247960f <unknown>\n#19 0x5fa8024d71f8 <unknown>\n#20 0x5fa8024d73d6 <unknown>\n#21 0x5fa8024e85e6 <unknown>\n#22 0x7cf987577aa4 <unknown>\n#23 0x7cf987604a34 __clone", "trace": "browser = <selenium.webdriver.remote.webdriver.WebDriver (session=\"888b1b4ae6e0b500159ea684bab74a1b\")>\n\n    @allure.title(\"Применение одного промокода дважды\")\n    def test_apply_same_promocode_twice(browser):\n        logging.info(\"Start test_apply_same_promocode_twice\")\n    \n        open_main_page(browser)\n        add_pizza_to_cart(browser, \"425\")\n        open_account_page(browser)\n        fill_auth_form(browser, \"rick deckard\", \"1234\")\n        click_element(browser, (By.XPATH, \"//button[contains(@name, 'login')]\"), \"Нажать на кнопку «Войти»\")\n        open_checkout_page(browser)\n        click_element(browser, (By.XPATH, \"//a[contains(@class, 'showcoupon')]\"), \"Нажать на кнопку «Нажмите для ввода купона»\")\n        input_text(browser, (By.XPATH, \"//input[contains(@id, 'coupon_code')]\"), \"GIVEMEHALYAVA\", \"Ввести валидный промокод в поле для промокода\")\n        click_element(browser, (By.XPATH, \"//button[contains(@name, 'apply_coupon')]\"), \"Нажать на кнопку «Применить купон»\")\n        click_element(browser, (By.XPATH, \"//a[contains(@class, 'showcoupon')]\"),\n                      \"Нажать на кнопку «Нажмите для ввода купона»\")\n        input_text(browser, (By.XPATH, \"//input[contains(@id, 'coupon_code')]\"), \"GIVEMEHALYAVA\",\n                   \"Ввести валидный промокод в поле для промокода\")\n>       click_element(browser, (By.XPATH, \"//button[contains(@name, 'apply_coupon')]\"),\n                      \"Нажать на кнопку «Применить купон»\")\n\ntest_apply_same_promocode_twice.py:26: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n..\\..\\src\\utils\\web_helpers\\browser_actions.py:22: in click_element\n    el.click()\n..\\..\\.venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\webelement.py:120: in click\n    self._execute(Command.CLICK_ELEMENT)\n..\\..\\.venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\webelement.py:573: in _execute\n    return self._parent.execute(command, params)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n..\\..\\.venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:454: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x0000020878B78440>\nresponse = {'status': 400, 'value': '{\"value\":{\"error\":\"element click intercepted\",\"message\":\"element click intercepted: Element ...unknown>\\\\n#21 0x5fa8024e85e6 \\\\u003Cunknown>\\\\n#22 0x7cf987577aa4 \\\\u003Cunknown>\\\\n#23 0x7cf987604a34 __clone\\\\n\"}}'}\n\n    def check_response(self, response: dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                try:\n                    value = json.loads(value_json)\n                    if isinstance(value, dict):\n                        if len(value) == 1:\n                            value = value[\"value\"]\n                        status = value.get(\"error\", None)\n                        if not status:\n                            status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                            message = value.get(\"value\") or value.get(\"message\")\n                            if not isinstance(message, str):\n                                value = message\n                                message = message.get(\"message\")\n                        else:\n                            message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.ElementClickInterceptedException: Message: element click intercepted: Element <button type=\"submit\" class=\"button\" name=\"apply_coupon\" value=\"...\">Применить купон</button> is not clickable at point (509, 282). Other element would receive the click: <div class=\"inner\">...</div>\nE         (Session info: chrome=138.0.7204.183); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#elementclickinterceptedexception\nE       Stacktrace:\nE       #0 0x5fa8024e92ca <unknown>\nE       #1 0x5fa801f90550 <unknown>\nE       #2 0x5fa801fe8f5c <unknown>\nE       #3 0x5fa801fe6dc2 <unknown>\nE       #4 0x5fa801fe4462 <unknown>\nE       #5 0x5fa801fe3b80 <unknown>\nE       #6 0x5fa801fd62ea <unknown>\nE       #7 0x5fa802007bc2 <unknown>\nE       #8 0x5fa801fd5c6a <unknown>\nE       #9 0x5fa802007d8e <unknown>\nE       #10 0x5fa80202d9e6 <unknown>\nE       #11 0x5fa802007993 <unknown>\nE       #12 0x5fa801fd3d6b <unknown>\nE       #13 0x5fa801fd5141 <unknown>\nE       #14 0x5fa8024ae2ab <unknown>\nE       #15 0x5fa8024b20b9 <unknown>\nE       #16 0x5fa802495139 <unknown>\nE       #17 0x5fa8024b2c68 <unknown>\nE       #18 0x5fa80247960f <unknown>\nE       #19 0x5fa8024d71f8 <unknown>\nE       #20 0x5fa8024d73d6 <unknown>\nE       #21 0x5fa8024e85e6 <unknown>\nE       #22 0x7cf987577aa4 <unknown>\nE       #23 0x7cf987604a34 __clone\n\n..\\..\\.venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\errorhandler.py:232: ElementClickInterceptedException"}, "steps": [{"name": "Открыть главную страницу сайта Pizzeria", "status": "passed", "start": 1757612884572, "stop": 1757612887945}, {"name": "Навести курсор на пиццу с ID 425", "status": "passed", "start": 1757612887946, "stop": 1757612888768}, {"name": "Нажать на кнопку «В корзину» для пиццы с ID 425", "status": "passed", "start": 1757612888769, "stop": 1757612888858}, {"name": "Открыть страницу аккаунта", "status": "passed", "start": 1757612888858, "stop": 1757612889423}, {"name": "Ввести имя пользователя", "status": "passed", "start": 1757612889423, "stop": 1757612889528}, {"name": "Ввести пароль", "status": "passed", "start": 1757612889529, "stop": 1757612889610}, {"name": "Нажать на кнопку «Войти»", "status": "passed", "start": 1757612889610, "stop": 1757612890019}, {"name": "Открыть страницу оформления заказа", "status": "passed", "start": 1757612890020, "stop": 1757612890439}, {"name": "Нажать на кнопку «Нажмите для ввода купона»", "status": "passed", "start": 1757612890439, "stop": 1757612890524}, {"name": "Ввести валидный промокод в поле для промокода", "status": "passed", "start": 1757612890524, "stop": 1757612891114}, {"name": "Нажать на кнопку «Применить купон»", "status": "passed", "start": 1757612891115, "stop": 1757612891194}, {"name": "Нажать на кнопку «Нажмите для ввода купона»", "status": "passed", "start": 1757612891194, "stop": 1757612891265}, {"name": "Ввести валидный промокод в поле для промокода", "status": "passed", "start": 1757612891265, "stop": 1757612891354}, {"name": "Нажать на кнопку «Применить купон»", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.ElementClickInterceptedException: Message: element click intercepted: Element <button type=\"submit\" class=\"button\" name=\"apply_coupon\" value=\"...\">Применить купон</button> is not clickable at point (509, 282). Other element would receive the click: <div class=\"inner\">...</div>\n  (Session info: chrome=138.0.7204.183); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#elementclickinterceptedexception\nStacktrace:\n#0 0x5fa8024e92ca <unknown>\n#1 0x5fa801f90550 <unknown>\n#2 0x5fa801fe8f5c <unknown>\n#3 0x5fa801fe6dc2 <unknown>\n#4 0x5fa801fe4462 <unknown>\n#5 0x5fa801fe3b80 <unknown>\n#6 0x5fa801fd62ea <unknown>\n#7 0x5fa802007bc2 <unknown>\n#8 0x5fa801fd5c6a <unknown>\n#9 0x5fa802007d8e <unknown>\n#10 0x5fa80202d9e6 <unknown>\n#11 0x5fa802007993 <unknown>\n#12 0x5fa801fd3d6b <unknown>\n#13 0x5fa801fd5141 <unknown>\n#14 0x5fa8024ae2ab <unknown>\n#15 0x5fa8024b20b9 <unknown>\n#16 0x5fa802495139 <unknown>\n#17 0x5fa8024b2c68 <unknown>\n#18 0x5fa80247960f <unknown>\n#19 0x5fa8024d71f8 <unknown>\n#20 0x5fa8024d73d6 <unknown>\n#21 0x5fa8024e85e6 <unknown>\n#22 0x7cf987577aa4 <unknown>\n#23 0x7cf987604a34 __clone\n\n", "trace": "  File \"C:\\Users\\Admin\\PycharmProjects\\PythonProject\\src\\utils\\web_helpers\\browser_actions.py\", line 22, in click_element\n    el.click()\n    ~~~~~~~~^^\n  File \"C:\\Users\\Admin\\PycharmProjects\\PythonProject\\.venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\webelement.py\", line 120, in click\n    self._execute(Command.CLICK_ELEMENT)\n    ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Admin\\PycharmProjects\\PythonProject\\.venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\webelement.py\", line 573, in _execute\n    return self._parent.execute(command, params)\n           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Admin\\PycharmProjects\\PythonProject\\.venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py\", line 454, in execute\n    self.error_handler.check_response(response)\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^\n  File \"C:\\Users\\Admin\\PycharmProjects\\PythonProject\\.venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\errorhandler.py\", line 232, in check_response\n    raise exception_class(message, screen, stacktrace)\n"}, "start": 1757612891354, "stop": 1757612892443}], "attachments": [{"name": "log", "source": "a57463d3-8d3b-4e95-8b5b-97af58778c2e-attachment.txt", "type": "text/plain"}, {"name": "stdout", "source": "b697a80a-4429-4745-b670-9121a1e00c9b-attachment.txt", "type": "text/plain"}], "start": 1757612884572, "stop": 1757612892484, "uuid": "4431e32c-89c2-4b29-a342-d44c96085d59", "historyId": "fe964ad8bf6f7059a4cf4722eba38670", "testCaseId": "fe964ad8bf6f7059a4cf4722eba38670", "fullName": "tests.checkout_page.test_apply_same_promocode_twice#test_apply_same_promocode_twice", "labels": [{"name": "parentSuite", "value": "tests.checkout_page"}, {"name": "suite", "value": "test_apply_same_promocode_twice"}, {"name": "host", "value": "DESKTOP-2SK90NU"}, {"name": "thread", "value": "13972-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.checkout_page.test_apply_same_promocode_twice"}], "titlePath": ["tests", "checkout_page", "test_apply_same_promocode_twice.py"]}